### DISCUSSION TITLE
igpu cannot be identify in ubuntu22.04

### DISCUSSION THREAD

**Discussion by zoulee24** [openvinotoolkit/openvino#26604] [Q&A]
system: ubuntu 22.04 server, not desktop
`Linux pw-ey 5.15.0-121-generic #131-Ubuntu SMP Fri Aug 9 08:29:53 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux`
hardware: 12600H
```bash
lspci | grep -i "VGA\|3D\|Graphics"
```
output: `00:02.0 VGA compatible controller: Intel Corporation Alder Lake-P Integrated Graphics Controller (rev 0c)`

how to install openvino-2024.3?
```bash
python3.11 -m pip install openvino==2024.3.0
```

how to query devices
```python
import openvino
core = openvino.Core()
print(core.available_devices)
```
output
```bash
['CPU']
````

dont know how to fix it, Unable to locate the issue, neeeeed help

**Comment by zoulee24** [comment]
Is there a driver that needs to be installed? or ...

**Comment by zoulee24** [comment]
need install client driver: https://dgpu-docs.intel.com/driver/client/overview.html

**Comment by zoulee24** [comment]
dont know why reboot or something happen, iGPU device disappear.

kernal: `6.8.0-40-generic`

gpu driver install(ps: from https://dgpu-docs.intel.com/driver/client/overview.html#)

```bash
wget -qO - https://repositories.intel.com/gpu/intel-graphics.key | \
  sudo gpg --yes --dearmor --output /usr/share/keyrings/intel-graphics.gpg
echo "deb [arch=amd64,i386 signed-by=/usr/share/keyrings/intel-graphics.gpg] https://repositories.intel.com/gpu/ubuntu jammy client" | \
  sudo tee /etc/apt/sources.list.d/intel-gpu-jammy.list
sudo apt update | sudo apt install -y \
  intel-opencl-icd intel-level-zero-gpu level-zero intel-level-zero-gpu-raytracing \
  intel-media-va-driver-non-free libmfx1 libmfxgen1 libvpl2 \
  libegl-mesa0 libegl1-mesa libegl1-mesa-dev libgbm1 libgl1-mesa-dev libgl1-mesa-dri \
  libglapi-mesa libgles2-mesa-dev libglx-mesa0 libigdgmm12 libxatracker2 mesa-va-drivers \
  mesa-vdpau-drivers mesa-vulkan-drivers va-driver-all vainfo hwinfo clinfo | sudo reboot
```

all necessary packages for the tutorial interface have been installed.

I'm not sure which packages that must be installed are not installed

**Comment by Ganesh-403** [comment]
@zoulee24 The fact that

```bash
lspci | grep -i "VGA\|3D\|Graphics"
```

shows your **Alder Lake-P iGPU** means the hardware is visible to the OS. The important part is why OpenVINO only reports:

```python
['CPU']
```

OpenVINO will only enumerate a `GPU` device if the **Intel GPU runtime (OpenCL/Level Zero)** is installed and the GPU plugin can initialize successfully. Simply installing the `openvino` Python package is not enough. :contentReference[oaicite:0]{index=0}

A few things I'd check:

1. **Verify the render device exists**

   ```bash
   ls -l /dev/dri
   ```

   You should normally see something like:

   ```
   card0
   renderD128
   ```

   If `renderD128` is missing, the Intel graphics driver hasn't initialized correctly.

2. **Check the GPU runtime**

   ```bash
   clinfo
   ```

   or

   ```bash
   sycl-ls
   ```

   If these don't list your Intel GPU, OpenVINO won't be able to use it either. :contentReference[oaicite:1]{index=1}

3. **Kernel messages**

   Since you mentioned the GPU later "disappeared", I'd look at:

   ```bash
   dmesg | grep -Ei "i915|xe|drm"
   ```

   Any firmware loading failures or driver initialization errors there would explain why only the CPU is detected.

4. **User permissions**

   Make sure your user is in the `render` group:

   ```bash
   groups
   ```

   If not:

   ```bash
   sudo usermod -aG render $USER
   ```

   then log out and back in. Intel's GPU configuration guide specifically calls this out for Linux. :contentReference[oaicite:2]{index=2}

One thing I noticed is that you installed a large number of graphics packages, but the packages themselves don't guarantee that the driver initialized successfully. If the iGPU disappeared after reboot, I'd be more inclined to investigate the kernel/driver state rather than install additional packages.

Could you also share the output of:

```bash
ls -l /dev/dri
clinfo
dmesg | grep -Ei "i915|xe|drm"
python3 -c "import openvino as ov; print(ov.Core().available_devices)"
```

That should make it possible to determine whether this is:

- a missing OpenCL/Level Zero runtime,
- a driver initialization problem,
- a permissions issue,
- or something specific to the kernel/graphics stack.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
